lib/ansible/modules/system/vdo.py

```diff
@@ -0,0 +1,762 @@
+              default is 0.25.  The special decimal values 0.25, 0.5,
+              and 0.75 can be used, as can any positive integer.
+        required: false
+    indexmode:
+        description:
+            - Specifies the index mode of the Albireo index.  The
+              default is 'dense', which has a deduplication window of
+              1 GB of index memory per 1 TB of incoming data,
+              requiring 10 GB of index data on persistent storage.
+              The 'sparse' mode has a deduplication window of 1 GB of
+              index memory per 10 TB of incoming data, but requires
+              100 GB of index data on persistent storage.
+        required: false
+    ackthreads:
+        description:
+            - Specifies the number of threads to use for
+              acknowledging completion of requested VDO I/O operations.
+              Valid values are integer values from 1 to 100 (lower
+              numbers are preferable due to overhead).  The default is
+              1.
+        required: false
+    biothreads:
+        description:
+            - Specifies the number of threads to use for submitting I/O
+              operations to the storage device.  Valid values are
+              integer values from 1 to 100 (lower numbers are
+              preferable due to overhead).  The default is 4.
+        required: false
+    cputhreads:
+        description:
+            - Specifies the number of threads to use for CPU-intensive
+              work such as hashing or compression.  Valid values are
+              integer values from 1 to 100 (lower numbers are
+              preferable due to overhead).  The default is 2.
+        required: false
+    logicalthreads:
+        description:
+            - Specifies the number of threads across which to
+              subdivide parts of the VDO processing based on logical
+              block addresses.  Valid values are integer values from
+              1 to 100 (lower numbers are preferable due to overhead).
+              The default is 1.
+        required: false
+    physicalthreads:
+        description:
+            - Specifies the number of threads across which to
+              subdivide parts of the VDO processing based on physical
+              block addresses.  Valid values are integer values from
+              1 to 16 (lower numbers are preferable due to overhead).
+              The physical space used by the VDO volume must be
+              larger than (slabsize * physicalthreads).  The default
+              is 1.
+        required: false
+notes:
+  - In general, the default thread configuration should be used.
+'''
+
+EXAMPLES = '''
+# Create a VDO volume
+- name: Create 2 TB VDO volume vdo1 on device /dev/md0
+  vdo:
+    name: vdo1
+    state: present
+    device: /dev/md0
+    logicalsize: 2T
+
+# Remove a VDO volume
+- name: Remove VDO volume vdo1
+  vdo:
+    name: vdo1
+    state: absent
+'''
+
+RETURN = '''# '''
```

Comment:
not a blocker: you return a lot of data, just documenting the top fields would be nice

Reply:
Is there documentation on how to document these fields?  There were previous test failures that were remnants of the "hello world" example in Ansible Module Development Walkthrough, which prompted me to remove the entire RETURN section, but ansible-test requires at least some return section.  I would like to avoid any "required key not provided" test failures, but I don't yet know what the test expects to see.

Reply:
no real docs, but you can look at other modules that do it, i.e copy

Reply:
The module now returns no values, except for "changed".

Reply:
All of the feedback to the user is now via module.log() calls.

Reply:
Is there anything else needed for this pull request to be accepted?

Reply:
FYI, this section of the code isn't going to change, because other parts were changed to remove the previous returned items.  Please re-review.

Reply:
Checking in after the end-of-year holiday; have you had a chance to re-review this?